src/linear_algebra/alternating.lean

```diff
@@ -434,3 +444,292 @@ lemma comp_multilinear_map_alternatization (g : N' →ₗ[R] N'₂)
+  (v : ιa ⊕ ιb → M) (i j : ιa ⊕ ιb) (hv : v i = v j) (hij : i ≠ j) :
+  dom_coprod_fun a b v = 0 :=
+begin
+  unfold dom_coprod_fun,
+  dsimp only,
+  rw multilinear_map.sum_apply,
+  apply finset.sum_involution
+    (λ σ _, (equiv.swap i j • σ : perm.mod_sum_congr ιa ιb))
+    (λ σ, _)
+    (λ σ, _)
+    (λ σ _, finset.mem_univ _)
+    (λ σ, _),
+  all_goals {
+    apply σ.induction_on' (λ σ, _),
+    rintro _, },
+  { dsimp only [quotient.lift_on'_mk', quotient.map'_mk', mul_action.quotient.smul_mk],
+    rw [perm.sign_mul, perm.sign_swap hij],
+    simp only [one_mul, units.neg_mul, function.comp_app, neg_smul, perm.coe_mul,
+      units.coe_neg, multilinear_map.smul_apply, multilinear_map.neg_apply,
+      multilinear_map.dom_dom_congr_apply, multilinear_map.dom_coprod_apply],
+    convert add_right_neg _;
+    { ext k, rw equiv.apply_swap_eq_self hv }, },
+  { dsimp only [quotient.lift_on'_mk', quotient.map'_mk', multilinear_map.smul_apply,
+      multilinear_map.dom_dom_congr_apply, multilinear_map.dom_coprod_apply],
+    apply mt,
+    intro hσ,
+    with_cases {
+      cases hi : σ⁻¹ i;
+        cases hj : σ⁻¹ j;
+        rw perm.inv_eq_iff_eq at hi hj;
+        propagate_tags' { substs hi hj, }, },
+    case [sum.inl sum.inr : i' j', sum.inr sum.inl : i' j'] {
+      -- the term pairs with and cancels another term
+      all_goals { obtain ⟨⟨sl, sr⟩, hσ⟩ := quotient.eq'.mp hσ, },
+      work_on_goal 0 { replace hσ := equiv.congr_fun hσ (sum.inl i'), },
+      work_on_goal 1 { replace hσ := equiv.congr_fun hσ (sum.inr i'), },
+      all_goals {
+        rw [←equiv.mul_swap_eq_swap_mul, mul_inv_rev, equiv.swap_inv, inv_mul_cancel_right] at hσ,
+        simpa using hσ, }, },
+    case [sum.inr sum.inr : i' j', sum.inl sum.inl : i' j'] {
+      -- the term does not pair but is zero
+      all_goals { convert smul_zero _, },
+      work_on_goal 0 { convert tensor_product.tmul_zero _ _, },
+      work_on_goal 1 { convert tensor_product.zero_tmul _ _, },
+      all_goals { exact alternating_map.map_eq_zero_of_eq _ _ hv (λ hij', hij (hij' ▸ rfl)), },
+      }, },
+  { exact _root_.congr_arg (quot.mk _) (equiv.swap_mul_involutive i j σ), }
+end
+
+/-- Like `multilinear_map.dom_coprod`, but ensures the result is also alternating.
+
+Note that this is usually defined over integer indices `ιa = fin n` and `ιb = fin m`, as
+$$
+(f \wedge g)(u_1, \ldots, u_{m+n}) =
+  \sum_{\operatorname{shuffle}(m, n)} \operatorname{sign}(\sigma)
+    f(u_{\sigma(1)}, \ldots, u_{\sigma(m)}) g(u_{\sigma(m+1)}, \ldots, u_{\sigma(m+n)}),
+$$
+where $\operatorname{shuffle}(m, n)$ consists of all permutations of $[1, m+n]$ such that
+$\sigma(1) < \cdots < \sigma(m)$ and $\sigma(m+1) < \cdots < \sigma(m+n)$.
+
+Here, we generalize this by replacing:
+* the product in the sum with a tensor product
+* the filtering of $[1, m+n]$ to shuffles with an isomorphic quotient
+* the additions in the subscripts of $\sigma$ with a index of type `sum`
+-/
+@[simps]
```

Comment:
```suggestion
```
As far as I understand, `dom_coprod_fun` is an implementation detail here, and you don't want `dom_coprod` to unfold into `dom_coprod_fun`.